libs/cache_line/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+#![feature(const_trait_impl)]
+#![no_std]
+
+use core::{mem, ops};
+
+const LINE_SIZE: usize = 64;
+
+/// A cache line optimised array.
```

Comment:
Also we should clearly state that we assume 64-byte cache lines in both the crate-level and here in this type's docs, or alternatively provide a way for the user to set a cache line size as a type parameter